docs/source/howto/run_codes.rst

```diff
@@ -357,24 +357,215 @@ After :ref:`setting up your computer <how-to:run-codes:computer>` and :ref:`sett
+    I.e. if the input of **C'** were new nodes **D1'** and **D2'** with the same content (hash) as **D1**, **D2**, the cache would trigger as well.
+
+
+.. note:: Caching is **not** implemented at the WorkChain/workfunction level (see :ref:`how-to:run-codes:caching:limitations` for details).
+
+
+.. _how-to:run-codes:caching:hashing:
+
+How are nodes hashed
+--------------------
+
+*Hashing* is turned on by default, i.e. all nodes in AiiDA are hashed.
+This means that even when you enable caching once you have already completed a number of calculations, those calculations can still be used retro-actively by the caching mechanism since their hashes have been computed.
+
+The hash of a ``Data`` node is computed from:
+
+* all attributes of the node, except the ``_updatable_attributes`` and ``_hash_ignored_attributes``
+* the ``__version__`` of the package which defined the node class
+* the content of the repository folder of the node
+* the UUID of the computer, if the node is associated with one
+
+The hash of a :class:`~aiida.orm.ProcessNode` includes, on top of this, the hashes of all of its input ``Data`` nodes.
+
+Once a node is stored in the database, its hash is stored in the ``_aiida_hash`` extra, and this extra is used to find matching nodes.
+If a node of the same class with the same hash already exists in the database, this is considered a cache match.
+Use the :meth:`~aiida.orm.nodes.Node.get_hash` method to check the hash of any node.
+In order to figure out why a calculation is *not* being reused, the :meth:`~aiida.orm.nodes.Node._get_objects_to_hash` method may be useful:
+
+.. code-block:: ipython
+
+    In [5]: node = load_node(1234)
+
+    In [6]: node.get_hash()
+    Out[6]: '62eca804967c9428bdbc11c692b7b27a59bde258d9971668e19ccf13a5685eb8'
+
+    In [7]: node._get_objects_to_hash()
+    Out[7]:
+    [
+        '1.0.0',
+        {
+            'resources': {'num_machines': 2, 'default_mpiprocs_per_machine': 28},
+            'parser_name': 'cp2k',
+            'linkname_retrieved': 'retrieved'
+        },
+        <aiida.common.folders.Folder at 0x1171b9a20>,
+        '6850dc88-0949-482e-bba6-8b11205aec11',
+        {
+            'code': 'f6bd65b9ca3a5f0cf7d299d9cfc3f403d32e361aa9bb8aaa5822472790eae432',
+            'parameters': '2c20fdc49672c3505cebabacfb9b1258e71e7baae5940a80d25837bee0032b59',
+            'structure': 'c0f1c1d1bbcfc7746dcf7d0d675904c62a5b1759d37db77b564948fa5a788769',
+            'parent_calc_folder': 'e375178ceeffcde086546d3ddbce513e0527b5fa99993091b2837201ad96569c'
+        }
+    ]
+
+
+.. _how-to:run-codes:caching:configure:
+
+How to configure caching
+------------------------
+
+Class level
+...........
+
+Besides an on/off switch per profile, the ``.aiida/cache_config.yml`` provides control over caching at the level of specific calculations using their corresponding entry point strings (see the output of ``verdi plugin list aiida.calculations``):
+
+.. code-block:: yaml
+
+    profile-name:
```

Comment:
Is `profile-name` a key-word or a placeholder for the *profile name*? I assume it's the latter, but this could be confusing.

Reply:
it should be the actual profile name. I assumed this was going to be clear because I explained this in "How to enable caching". I can repeat the extra clarify I added there as well

Reply:
Since users are expected to jump around in the how-tos, I think that duplication is warranted. Also, I missed it, and I probably already read this more thoroughly then most readers would. ;)